### DISCUSSION TITLE
“ln -s”+WebDAV曲线救国实现Alist作为WordPress的外联图床。

### DISCUSSION THREAD

**Discussion by cat9999sss** [AlistGo/alist#2547] [Ideas]
Alist作为一款非常不错的网盘程序，很多功能让我觉得做的特别出色。尤其是AList 支持所有WebDAV 存储。
<p id="webdav-配置" tabindex="-1">简单的说一下我们要做的：</p>
<p tabindex="-1">在服务器上挂载WebDAV,建立图床目录并与wordpress的upload目录软连。</p>
安装所需程序：
Ubuntu：
<pre>sudo apt-get install davfs2 -y
</pre>
CentOS：
<pre>sudo yum install davfs2 -y
</pre>
创建挂载目录：
<pre>sudo mkdir /mnt/WebDAV
</pre>
挂载WebDAV服务到本地目录：
<pre>sudo mount -t davfs -o noexec https://example.com/webdav/ /mnt/WebDAV/
</pre>
之后会要求输入账户和密码登信息。挂载成功后，即可当正常磁盘一样访问WebDAV服务了。速度快慢取决于你自身和服务商的网速。

解除挂载方法：
<pre>sudo umount /mnt/WebDAV
</pre>
使用fstab挂载WebDAV：
<pre>$ cat &lt;&lt; EOF | sudo tee -a /etc/fstab
# personal webdav
https://example.com/webdav/ /mnt/WebDAV davfs _netdev,noauto,user,uid=nobody,gid=nobody 0 0
EOF
</pre>
保存账户密码：
<pre>cat &lt;&lt; EOF | sudo tee -a /etc/davfs2/secrets
/mnt/dav account password
EOF
</pre>
之后在挂载的目录中新建一个图床文件夹。然后做一个软链。例如
<pre>ln -s /源文件夹 /目标文件夹 
</pre>
我们是给wordpress的upload文件夹做软件，所以是：
<pre>ln -s /www/wwwroot/dashen.wang/wp-content/uploads /mnt/WebDAV/uploads
</pre>

做完这些后，你会发现wordpress上传的文件直接上传到你的网盘里了。既做了备份，又解决了储存空间的问题，再也不怕上传大文件心疼硬盘空间了。

喜欢的朋友可以支持一下我 [https://dashen.wang/833.html](原文地址：https://dashen.wang/833.html)

**Comment by cat9999sss** [comment]
以上都是脑补的想法，有人试了之后来回复一下我啊。哈哈哈哈。
注意可能遇到写入的权限问题。因为挂载WebDAV的操作都是sudo的。

**Reply by shaojs321** [reply]
技术不是问题，问题是频繁调佣网盘的api是否会被封

**Reply by cat9999sss** [reply]
好像是这样的。

**Reply by iwvw** [reply]
不错，抽空试试
就是不知道性能怎么样

**Reply by xiaofami** [reply]
不错，我来补充自动挂载webdav的Systemd文件：

```bash
# /etc/systemd/system/mnt-WebDAV.mount
[Unit]
Description=Mount alist webdav share
After=network-online.target
Wants=network-online.target

[Mount]
What=https://example.com/webdav/
Where=/mnt/WebDAV
Options=noauto,user,uid=1000,gid=1000
Type=davfs
TimeoutSec=60

[Install]
WantedBy=remote-fs.target
```
```bash
# /etc/systemd/system/mnt-WebDAV.automount
[Unit]
Description=Mount alist webdav share
After=network-online.target
Wants=network-online.target

[Automount]
Where=/mnt/WebDAV
TimeoutIdleSec=300

[Install]
WantedBy=remote-fs.target
```
修改 /etc/davfs2/secrets 文件，加入 

```
https://example.com/webdav/ 用户名 密码
```
 一行。然后启用服务即可。我唯一的顾虑是，通过davfs2 挂载后，alist是否可以正确实现302重定向使得客户端获取文件真实地址？因为我想在一个百兆网口的机顶盒上搭建alist和minidlna服务，如果客户端无法获取真实地址百兆网口就会严重影响使用体验。

**Reply by xiaofami** [reply]
**alist - webdav - davfs2 - minidlna** 方法行不通，用VLC测试能连接到DLNA服务器，也能看到部分目录，但是所有目录都是空的。本地硬盘通过DLNA共享就一切正常。推测是dlna的timeout机制导致文件列表没有获取完全，说到底dlna本来就是服务于本地文件，不适合远程挂载。

**Comment by tcpqueue** [comment]
这边建议使用rclone挂载，rclone有缓存机制，我是这样的，另外用ln -s要关闭网站防跨站，不然用不了
